src/physics/cam/gw_ml.F90

```diff
@@ -0,0 +1,419 @@
+module gw_ml
+
+!
+! This module handles machine learnt gravity wave schemes developed by
+! the DataWave Project
+!
+
+use gw_utils, only: r8
+use ppgrid,   only: pver
+use spmd_utils,      only: mpicom, mstrid=>masterprocid, masterproc, mpi_real8
+use cam_abortutils, only: endrun
+use cam_logfile,    only: iulog
+
+use ftorch
+
+implicit none
+private
+save
+
+public :: gw_drag_convect_dp_ml, gw_drag_convect_dp_ml_init, gw_drag_convect_dp_ml_final
+
+! Neural Net as read in by FTorch
+type(torch_model) :: convect_net
+
+! Means for normalisation
+real(r8) :: utgw_mean(pver), vtgw_mean(pver)
+real(r8) :: u_mean(pver), v_mean(pver)
+real(r8) :: t_mean(pver)
+real(r8) :: dse_mean(pver)
+real(r8) :: nm_mean(pver)
+real(r8) :: netdt_mean(pver)
+real(r8) :: zm_mean(pver)
+real(r8) :: rhoi_mean(pver+1)
+real(r8) :: ps_mean
+real(r8) :: lat_mean
+real(r8) :: lon_mean
+! Standard deviations for normalisation
+real(r8) :: utgw_std(pver), vtgw_std(pver)
+real(r8) :: u_std(pver), v_std(pver)
+real(r8) :: t_std(pver)
+real(r8) :: dse_std(pver)
+real(r8) :: nm_std(pver)
+real(r8) :: netdt_std(pver)
+real(r8) :: zm_std(pver)
+real(r8) :: rhoi_std(pver+1)
+real(r8) :: ps_std
+real(r8) :: lat_std
+real(r8) :: lon_std
+
+contains
+
+!==========================================================================
+
+subroutine gw_drag_convect_dp_ml(ncol, dt, &
+                                 u, v, t, dse, nm, netdt, zm, rhoi, ps, lat, lon, &
+                                 utgw, vtgw)
+
+  ! Take data from CAM, normalise and concatenate before passing it to the Torch neural
+  ! net to calculate u and v tendencies.
+
+
+
+
+  ! Column dimension.
```

Comment:
I think this routine is missing some runtime checks on expected input shapes. 

Could we validate the data dimensions of data coming from CAM? i.e. sizes and shapes are consistent with what we expect ncol to be, for example. Perhaps these checks are done elsewhere?

Reply:
It is being passed from the main CAM as an `intent(in)` so I'm not quite sure I understand.

Reply:
OK, ignore this request. I was thinking of adding some runtime checks on shape or physical consistency - but given this has already passed throughg CAM  it probably isn't necessary here. 

Should we add any checks on